pkg/apis/nfd/nodefeaturerule/rule_test.go

```diff
@@ -54,7 +54,7 @@ func TestRule(t *testing.T) {
 	assert.Equal(t, r1.Labels, m.Labels, "empty matcher should have matched empty features")
 
 	_, err = Execute(r2, f)
-	assert.Error(t, err, "matching against a missing feature should have returned an error")
+	assert.NoError(t, err, "matching against a missing feature should not have returned an error")
```

Comment:
We should add similar asserts to the result as we did for the other test case below. Also, could you squash the commits into one?

Reply:
Done!